corehq/motech/openmrs/models.py

```diff
@@ -37,6 +39,26 @@ class ColumnMapping(DocumentSchema):
     property = StringProperty()
     data_type = StringProperty(choices=DATA_TYPES, required=False)
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._importer = None
+
+    def set_importer(self, value):
+        self._importer = value
```

Comment:
Interesting use of a setter method. Why not use a plain attribute and assignment?

```py
def __init__(...)
    ...
    self.importer = None
```
and later
```py
mapping.importer = importer
```

Reply:
It's a jsonobject, so even though `importer` is an instance attribute, it'll still get returned by `to_json()`:
```python
In [12]: class Foo(DocumentSchema):
    ...:     foo = StringProperty()
    ...:     def __init__(self, *args, **kwargs):
    ...:         super().__init__(*args, **kwargs)
    ...:         self.bar = None
    ...:         

In [13]: foo = Foo()

In [14]: foo.to_json()
Out[14]: {'bar': None, 'doc_type': 'Foo', 'foo': None}
```

`ColumnMapping` instances are rendered in a JSON UI, so we would have to filter out the attributes we don't want users to see (or only select the ones we want them to see) in the view. It's messy. I figured a setter, although clumsy, would be less clumsy than the alternative.